docs/wiki/Solution creation.md

```diff
@@ -692,3 +718,74 @@ jobs:
 ```
 
 </details>
+
+# General solution creation
+
+When creating a solution that leverages CARML modules, there are several aspects to consider. This sub-section intends to provide you with a rough step-by-step guide to get you started.
+
+1. Identify the resources & deployment scope needed for your architecture
+
+    When you want to create your solution you should first gain an understanding of its designated architecture and and a results its inherently required services. For this and the subsequent steps, let's consider the following scenario:
+
+    - You want to deploy a Virtual Machine that is able to connect privately to a storage account
+    - For this architecture you may use the following services
+      - 1 resource group to place your resources in
+      - 1 Network Security Group to allow traffic control for your subnets
+      - 1 Virtual Network with 2 subnets
+      - 1 Storage Account with 1 Private Endpoint that connects into the 1st Virtual Network Subnet
+      - 1 Virtual Machine that is deployed into the 2nd Virtual Network subnet
+
+    Also, you need to consider the scope you want to deploy into. In the above example, we want to deploy a Resource Group, which must be deployed into a Subscription scope. All other resources, in turn can be deployed into the resource group scope of that resource group.
+
+    <p>
+
+1. Identify dependencies between them
+
+    Next, you need to know in which order you need to deploy those resources. For example, as all resources must be placed in a resource group, the resource group must be deployed first. Likewise, before you can deploy a Virtual Machine, you first need to create a Virtual Network. All together this may look like
+
+      ::: mermaid
+      graph LR;
+      rg[Resource Group]
+      vnet[Virtual Network]
+      st[Storage Account]
+      pe[Private Endpoint]
+      vm[Virtual Machine]
+      nsg[Network Security Group]
+
+
+      rg --> vnet
+      rg --> st
+      rg --> vm
+      rg --> nsg
+      vnet --> pe
+      st --> pe
+      nsg --> vnet
+      vnet --> vm
+      :::
+
+1. Consider orchestration options
+
+    With the services & dependencies identified, the next question is, how those dependencies can be implemented. As described in the [Orchestration Overview](#orchestration-overview) sub-section, this is primarily a decision about 'pipeline-orchestration' vs. 'template-orchestration' - and in case of the later, if there are any steps that have to run in the pipeline regardless (for example an upload of files).
```

Comment:
```suggestion
    With the services & dependencies identified, the next question is, how those dependencies can be implemented. As described in the [Orchestration Overview](#orchestration-overview) sub-section, this is primarily a decision about 'pipeline-orchestration' vs. 'template-orchestration' - and in case of the latter, if there are any steps that have to run in the pipeline regardless (for example an upload of files).
```